isisd/isis_cli.c

```diff
@@ -1596,93 +1596,118 @@ void cli_show_isis_sr_enabled(struct vty *vty, struct lyd_node *dnode,
 }
 
 /*
- * XPath: /frr-isisd:isis/instance/segment-routing/srgb
+ * XPath: /frr-isisd:isis/instance/segment-routing/label-block
  */
-DEFPY_YANG (isis_sr_global_block_label_range,
-       isis_sr_global_block_label_range_cmd,
-       "segment-routing global-block (16-1048575)$lower_bound (16-1048575)$upper_bound",
-       SR_STR
-       "Segment Routing Global Block label range\n"
-       "The lower bound of the block\n"
-       "The upper bound of the block (block size may not exceed 65535)\n")
+
+DEFPY_YANG(
+	isis_sr_global_block_label_range, isis_sr_global_block_label_range_cmd,
+	"segment-routing global-block (16-1048575)$gb_lower_bound (16-1048575)$gb_upper_bound [local-block (16-1048575)$lb_lower_bound (16-1048575)$lb_upper_bound]",
+	SR_STR
+	"Segment Routing Global Block label range\n"
+	"The lower bound of the global block\n"
+	"The upper bound of the global block (block size may not exceed 65535)\n"
+	"Segment Routing Local Block label range\n"
+	"The lower bound of the local block\n"
+	"The upper bound of the local block (block size may not exceed 65535)\n")
 {
-	nb_cli_enqueue_change(vty, "./segment-routing/srgb/lower-bound",
-			      NB_OP_MODIFY, lower_bound_str);
-	nb_cli_enqueue_change(vty, "./segment-routing/srgb/upper-bound",
-			      NB_OP_MODIFY, upper_bound_str);
+	nb_cli_enqueue_change(vty,
+			      "./segment-routing/label-blocks/srgb/lower-bound",
+			      NB_OP_MODIFY, gb_lower_bound_str);
+	nb_cli_enqueue_change(vty,
+			      "./segment-routing/label-blocks/srgb/upper-bound",
+			      NB_OP_MODIFY, gb_upper_bound_str);
+
+	if (lb_lower_bound) {
+		nb_cli_enqueue_change(vty,
+			      "./segment-routing/label-blocks/srlb/lower-bound",
+			      NB_OP_MODIFY, lb_lower_bound_str);
+		nb_cli_enqueue_change(vty,
+			      "./segment-routing/label-blocks/srlb/upper-bound",
+			      NB_OP_MODIFY, lb_upper_bound_str);
+	}
```

Comment:
Do you handle this scenario:
`segment-routing global-block 20000 23999 local-block 15000 15999`
Then
`segment-routing global-block 20000 23999`

In this case, the srlb should be reset to its default value like with the `no segment-routing ...`

Reply:
I've changed the command so that it behaves in this way